interfaces/dotnet/Cantera/src/CanteraException.cs

```diff
@@ -0,0 +1,14 @@
+using Cantera.Interop;
+
+namespace Cantera;
+
+public class CanteraException : InvalidOperationException
```

Comment:
`CanteraError` is kind of a catch-all, so would it make sense for this to derive from a more generic error type? I'll admit that I'm not familiar with the C# exception hierarchy.

Reply:
That is an astute observation. It should/can use a more generic base class, probably `System.Exception` itself.

(Note: In C# it's generally discouraged to throw or catch [outside of your entrypoint] `System.Exception` itself, instead preferring a more-derived exception)